src/main/java/pacApp/pacController/UserController.java

```diff
@@ -0,0 +1,115 @@
+package pacApp.pacController;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import pacApp.pacData.UserRepository;
+import pacApp.pacModel.Currency;
+import pacApp.pacModel.User;
+import pacApp.pacModel.request.Booking;
+import pacApp.pacModel.request.UserInfo;
+import pacApp.pacModel.response.GenericResponse;
+import pacApp.pacSecurity.JwtAuthenticatedProfile;
+
+import java.util.Optional;
+
+@RestController
+public class UserController {
+
+    private static final Logger log = LoggerFactory.getLogger(AuthenticationController.class);
+    private final UserRepository repository;
+
+    public UserController(UserRepository repository) {
+        this.repository = repository;
+    }
+
+    @RequestMapping(value = "/user", method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getUserInfo() {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+
+        if (!(auth instanceof JwtAuthenticatedProfile)) {
+            GenericResponse response = new GenericResponse(HttpStatus.FORBIDDEN.value(),"Authentication failure");
+            return new ResponseEntity<>(response, HttpStatus.FORBIDDEN);
+        }
+
+        JwtAuthenticatedProfile authenticatedProfile = (JwtAuthenticatedProfile) auth;
+        String userEmail = authenticatedProfile.getName();
+
+        Optional<User> optUser = this.repository.findOneByEmail(userEmail);
+
+        if (!optUser.isPresent()){
+            GenericResponse response = new GenericResponse(HttpStatus.BAD_REQUEST.value(),"Invalid user");
+            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
+        }
+
+        User user = optUser.get();
+
+        UserInfo userInfo = this.convertUserToUserInfo(user);
+
+        GenericResponse response = new GenericResponse(HttpStatus.OK.value(),"Invalid user");
+        return new ResponseEntity<>(userInfo, HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/user", method = RequestMethod.PUT,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity updateUser(@RequestBody UserInfo userInfo) {
+        if (userInfo == null) {
+            GenericResponse response = new GenericResponse(400, "Incorrect rental id");
```

Comment:
Rental Id? don't you mean user info?

Reply:
My fault. Fixed in commit [98dba93](https://github.com/JamHaven/se-carrental-webservice/pull/20/commits/98dba93dcd164b86e9b8db3fb636780f53ac6b2c)